src/software/util/parameter/parameter.h

```diff
@@ -3,17 +3,16 @@
 #include <functional>
 #include <memory>
 #include <mutex>
+#include <optional>
 #include <string>
 #include <vector>
 
 /**
  * This class defines a dynamic parameter, meaning the parameter
- * value can be changed during runtime. Although the class is templated, it is only meant
- * to support the types also supported by the ROS Parameter Server.
+ * value can be changed during runtime.
  *
- * See http://wiki.ros.org/Parameter%20Server for the list of types
+ * In our codebase, we currently support bool, int32_t, double, and strings
```

Comment:
why `int32_t` over `int`?

Reply:
it was just a ROS quirk, we just use `int` now, updated the comment